aloha-core/src/main/scala/com/eharmony/aloha/models/multilabel/MultilabelModel.scala

```diff
@@ -0,0 +1,387 @@
+package com.eharmony.aloha.models.multilabel
+
+import java.io.{Closeable, PrintWriter, StringWriter}
+
+import com.eharmony.aloha.audit.Auditor
+import com.eharmony.aloha.dataset.density.Sparse
+import com.eharmony.aloha.factory._
+import com.eharmony.aloha.id.ModelIdentity
+import com.eharmony.aloha.models._
+import com.eharmony.aloha.models.reg.RegressionFeatures
+import com.eharmony.aloha.reflect.{RefInfo, RefInfoOps}
+import com.eharmony.aloha.semantics.Semantics
+import com.eharmony.aloha.semantics.func.{GenAggFunc, GenAggFuncAccessorProblems}
+import spray.json.{JsonFormat, JsonReader}
+
+import scala.collection.{immutable => sci, mutable => scm}
+import scala.util.{Failure, Success, Try}
+
+
+// TODO: When adding label-dep features, a Seq[GenAggFunc[K, Sparse]] will be needed.
+// TODO: To create a Seq[GenAggFunc[K, Sparse]], a Semantics[K] needs to be derived from a Semantics[A].
+// TODO: MorphableSemantics provides this.  If K is *embedded inside* A, it should be possible in some cases.
+// TODO: An alternative is to pass a Map[K, Sparse], Map[K, Option[Sparse]], Map[K, Seq[Sparse]] or something.
+// TODO: Directly passing the map of LDFs avoids the need to derive a Semantics[K].  This is easier to code.
+// TODO: Directly passing LDFs would however be more burdensome to the data scientists.
+
+/**
+  * A multi-label predictor.
+  *
+  * Created by ryan.deak on 8/29/17.
+  *
+  * @param modelId An identifier for the model.  Used in score and error reporting.
+  * @param featureNames feature names (parallel to featureFunctions)
+  * @param featureFunctions feature extracting functions.
+  * @param labelsInTrainingSet a sequence of all labels encountered during training. Note: the
+  *                            order of labels may relate to the predictor produced by
+  *                            predictorProducer.  It is the caller's responsibility to ensure
+  *                            the order is correct.  To mitigate such problems, both labels
+  *                            and indices into labelsInTrainingSet are passed to the predictor
+  *                            produced by predictorProducer.
+  * @param labelsOfInterest if provided, a sequence of labels will be extracted from the example
+  *                         for which a prediction is desired. The ''intersection'' of the
+  *                         extracted labels and the training labels will be the labels for which
+  *                         predictions will be produced.
```

Comment:
I think we should add something like `If a provided label is not in the training label set it is ignored`.